src/main/java/com/pub/wysio/controller/CustomerController.java

```diff
@@ -0,0 +1,49 @@
+package com.pub.wysio.controller;
+
+import com.pub.wysio.entity.Customer;
+import com.pub.wysio.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class CustomerController {
+    @Autowired
+    private CustomerService service;
+
+    @PostMapping("/addCustomer")
+    public Customer addCustomer(@RequestBody Customer customer) {
+        return service.saveCustomer(customer);
+    }
+
+    @PostMapping("/addCustomers")
+    public List<Customer> addCustomers(@RequestBody List<Customer> customers) {
+        return service.saveCustomers(customers);
+    }
+
+    @GetMapping("/customers")
+    public List<Customer> findAllCustomers() {
+        return service.getAllCustomers();
+    }
+
+    @GetMapping("/customerById/{id}")
+    public Customer findCustomerById(@PathVariable int id) {
+        return service.getCustomerById(id);
+    }
+
+    @GetMapping("/customer/{nickname}")
+    public Customer findCustomerByNickname(@PathVariable String nickname) {
```

Comment:
We cannot really utilise this endpoint, because the nickname is not unique. If we have more than one customer with the same nickname in the db, the api will probably return the first object every time.